server/api/controllers/companies.js

```diff
@@ -0,0 +1,180 @@
+var mongoose = require('mongoose');
+var Company = require('../models/company');
+var User = require('../models/user');
+var userAPI = require('./users');
+var _ = require('lodash');
+
+var companyAPI = {
+  companyObj: {
+    company: {},
+    user: {}
+  },
+
+  createCompanyObj: function(req){
+    var companyObject = new Company({
+      companyName: req.body.company.companyName,
+      isParent: req.body.company.isParent,
+      address: {
+        address1: req.body.company.address.address1,
+        address2: req.body.company.address.address2,
+        city: req.body.company.address.city,
+        state: req.body.company.address.state,
+        country: req.body.company.address.country,
+        zip: req.body.company.address.zip
+      },
+      clients: req.body.company.clients,
+      phone: req.body.company.phone,
+      fax: req.body.company.fax,
+      cell: req.body.company.cell,
+      email: req.body.company.email,
+      created: Date.now()
+    });
+
+    return companyObject;
+  },
+
+  createCompany: function(companyObj, userObj, parentCompanyId){
+    var self = this;
+
+    companyObj.save(function(err){
+      if (err) throw new Error('Company could not be created becase of ' + err.message);
+      Company.findById(companyObj, function(err, doc){
+        if (err) throw new Error('Company could not be created becase of ' + err.message);
+        if (parentCompanyId){
+          self.addClienToCompany(parentCompanyId, doc._id);
+        }
+      });
+    });
+
+    userAPI.createUser(userObj, function(err, user){
+      if (err) {
```

Comment:
We will not be relying on the completion of adding a user for the completion of the company. We should take this out of this area.